src/System.Text.Json/src/System/Text/Json/Node/JsonObject.cs

```diff
@@ -0,0 +1,494 @@
+        ///   <see langword="false"/> otherwise.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   Provided property name is null.
+        /// </exception>
+        public bool ContainsProperty(string propertyName) => propertyName != null ? _dictionary.ContainsKey(propertyName) : throw new ArgumentNullException(nameof(propertyName));
+
+        /// <summary>
+        ///   Returns the value of a property with the specified name.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to return.</param>
+        /// <returns>Value of the property with the specified name.</returns>
+        /// <exception cref="KeyNotFoundException">
+        ///   Property with specified name is not found in JSON object.
+        /// </exception>
+        public JsonNode GetPropertyValue(string propertyName)
+        {
+            if (!TryGetPropertyValue(propertyName, out JsonNode jsonNode))
+            {
+                throw new KeyNotFoundException(string.Format(SR.PropertyNotFound, propertyName));
+            }
+
+            return jsonNode;
+        }
+
+        /// <summary>
+        ///   Returns the value of a property with the specified name.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to return.</param>
+        /// <param name="jsonNode">Value of the property with specified name.</param>
+        /// <returns>
+        ///  <see langword="true"/> if property with specified name was found;
+        ///  otherwise, <see langword="false"/>
+        /// </returns>
+        /// <remarks>
+        ///   When returns <see langword="false"/>, the value of <paramref name="jsonNode"/> is meaningless.
+        ///   Null <paramref name="jsonNode"/> doesn't mean the property value was "null" unless <see langword="true"/> is returned.
+        /// </remarks>
+        public bool TryGetPropertyValue(string propertyName, out JsonNode jsonNode) => _dictionary.TryGetValue(propertyName, out jsonNode);
+
+        /// <summary>
+        ///   Returns the JSON object value of a property with the specified name.
+        /// </summary>
+        /// <param name="propertyName">Name of the property to return.</param>
+        /// <returns>JSON objectvalue of a property with the specified name.</returns>
+        /// <exception cref="KeyNotFoundException">
+        ///   Property with specified name is not found in JSON object.
+        /// </exception>
+        /// <exception cref="InvalidCastException">
+        ///   Property with specified name is not a JSON object.
+        /// </exception>
+        public JsonObject GetJsonObjectPropertyValue(string propertyName)
+        {
+            if (GetPropertyValue(propertyName) is JsonObject jsonObject)
+            {
+                return jsonObject;
+            }
+
+            throw new InvalidCastException(string.Format(SR.PropertyTypeMismatch, propertyName));
```

Comment:
Don't use `string.Format`, but rather `SR.Format`. Here/elsewhere.

```suggestion
            throw new InvalidCastException(SR.Format(SR.PropertyTypeMismatch, propertyName));
```

Example:
https://github.com/dotnet/corefx/blob/f0ec4a345fc31bc9d075fec86fa6fb3ec88d43a2/src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs#L1156-L1159

https://github.com/dotnet/corefx/blob/f0ec4a345fc31bc9d075fec86fa6fb3ec88d43a2/src/Common/src/System/SR.cs#L44-L52